src/sentry/middleware/auth.py

```diff
@@ -43,9 +45,29 @@ def process_request(self, request):
         # If there is a valid signature on the request we override the
         # user with the user contained within the signature.
         user = process_signature(request)
+        auth = get_authorization_header(request).split()
+
         if user is not None:
             request.user = user
             request.user_from_signed_request = True
+        elif auth and auth[0].lower() == TokenAuthentication.token_name:
+            try:
+                result = TokenAuthentication().authenticate(request=request)
```

Comment:
If `result is None`, should `request.user = AnonymousUser()`?

Reply:
hmm yea! good catch

Reply:
When possible you should avoid raising exceptions inside a `try` just to have it caught in the `except`. Also excepting on `Exception` can mask bugs later on.
```py
try:
    result = TokenAuthentication().authenticate(request=request)
except AuthenticationFailed:
    result = None

if result:
    request.user, request.auth = result
else:
    # default to anonymous user and use IP ratelimit
    request.user = SimpleLazyObject(lambda: get_user(request))
```